docs: Improve trace_oomkill guide (#3493)
inspektor-gadget/inspektor-gadget: gadgets/trace_oomkill/README.mdx

<Tabs groupId="env">
    <TabItem value="kubectl-gadget" label="kubectl gadget">
        Kubernetes allows you to set memory limits on containers. When a container exceeds its memory limit, the kernel's out-of-memory (OOM) killer is invoked, which selects a process to kill to free up memory.
        Normally Kubernetes allows to check for OOM events only at the pod level so if a process is killed by the OOM killer, it is not possible to know which process was killed especially if the container has multiple processes running.
        This gadget allows you to trace the OOM kill events at process level. To demonstrate this, start by creating a deployment in a namespace:

        :::note
        The Kubernetes version used in this guide is `1.27.0` since `1.28.0` [introduced cgroup aware OOM killing](https://github.com/kubernetes/kubernetes/pull/117793) when possible.
        But even with `1.28.0+`, the OOM kill event might be missed if [cgroup v2 isn't enabled](https://kubernetes.io/docs/concepts/architecture/cgroups/#using-cgroupv2) or perhaps [singleProcessOOMKill](https://github.com/kubernetes/kubernetes/pull/126096) is used.

>         :::

Reviewer: I suggest putting this info together. Something like:

```
Before Kubernetes version `1.28.0`, the OOM kill events were only generated if the main container process (PID 1) was OOM killed. 
If any other process in the container was OOM killed, it would remain "invisible" to Kubernetes, meaning users wouldn't be notified. 
However, even after version `1.28.0`, OOM kill events will not be generated if [cgroup v2 isn't enabled](https://kubernetes.io/docs/concepts/architecture/cgroups/#using-cgroupv2)
or if [singleProcessOOMKill](https://github.com/kubernetes/kubernetes/pull/126096) is being used.

To ensure you are notified whenever any process within a container is OOM killed, regardless of whether it is PID 1 or not, you can use the `trace_oomkill` gadget. 
To demonstrate it, let's start by creating a deployment in a namespace:

:::note
This guide uses Kubernetes version `1.27.0` as an example of scenario were Kubernetes doesn't generate OOM kill events for processes differente than PID 1.
:::
```

        command terminated with exit code 137
        ```

        First check if pod was restarted:

        ```bash
        $ kubectl get pod -l app=oomkill-demo -n oomkill-demo
        NAME                           READY   STATUS    RESTARTS   AGE
        oomkill-demo-f5976f447-pr486   1/1     Running   0          70s

        Now, check if the OOM kill event was reported by Kubernetes:

        ```bash
        $ kubectl get events --field-selector=reason=OOMKilled -n oomkill-demo
        No resources found in oomkill-demo namespace.
        ```

        or check the `lastState` of the pod:

        ```bash
        $ kubectl get pod -l app=oomkill-demo -n oomkill-demo -o jsonpath="{.items[*].status.containerStatuses[*].lastState}"
        {}
        ```

>         Since the OOM kill event is not reported by Kubernetes, we can use the gadget to trace the OOM kill event at process level. So, go back to *the first terminal* and see:

Reviewer: I suggest no to use the terminology "pod level" and "process level", but something simpler:

```suggestion
        Even if the OOM kill event is not reported by Kubernetes, it's still notified by the `trace_oomkill` gadget. So, go back to *the first terminal* and see:
```
